dace/libraries/standard/nodes/reduce.py

```diff
@@ -1,870 +1,870 @@
-        e.data.dynamic = True
-
-        ### set reduce_node axes to all (needed)
-        reduce_node.axes = None
-
-        # fill scope connectors, done.
-        sdfg.fill_scope_connectors()
-
-        # finally, change the implementation to cuda (block)
-        # itself and expand again.
-        reduce_node.implementation = 'CUDA (block)'
-        sub_expansion = ExpandReduceCUDABlock(0, 0, {}, 0)
-        return sub_expansion.expansion(node=node, state=state, sdfg=sdfg)
-        #return reduce_node.expand(sdfg, state)
-
-
-@dace.library.node
-class Reduce(dace.sdfg.nodes.LibraryNode):
-    """ An SDFG node that reduces an N-dimensional array to an
-        (N-k)-dimensional array, with a list of axes to reduce and
-        a reduction binary function. """
-
-    # Global properties
-    implementations = {
-        'pure': ExpandReducePure,
-        'OpenMP': ExpandReduceOpenMP,
-        'CUDA (device)': ExpandReduceCUDADevice,
-        'CUDA (block)': ExpandReduceCUDABlock,
-        'CUDA (block allreduce)': ExpandReduceCUDABlockAll
-        # 'CUDA (warp)': ExpandReduceCUDAWarp,
-        # 'CUDA (warp allreduce)': ExpandReduceCUDAWarpAll
-    }
-
-    default_implementation = 'pure'
-
-    # Properties
-    axes = ListProperty(element_type=int, allow_none=True)
-    wcr = LambdaProperty(default='lambda a, b: a')
-    identity = Property(allow_none=True)
-
-    def __init__(self,
-                 wcr='lambda a, b: a',
-                 axes=None,
-                 identity=None,
-                 schedule=dtypes.ScheduleType.Default,
-                 debuginfo=None,
-                 **kwargs):
-        super().__init__(name='Reduce', **kwargs)
-        self.wcr = wcr
-        self.axes = axes
-        self.identity = identity
-        self.debuginfo = debuginfo
-        self.schedule = schedule
-
-    @staticmethod
-    def from_json(json_obj, context=None):
-        ret = Reduce("lambda a, b: a", None)
-        dace.serialize.set_properties_from_json(ret, json_obj, context=context)
-        return ret
-
-    def __str__(self):
-        # Autodetect reduction type
-        redtype = detect_reduction_type(self.wcr)
-        if redtype == dtypes.ReductionType.Custom:
-            wcrstr = unparse(ast.parse(self.wcr).body[0].value.body)
-        else:
-            wcrstr = str(redtype)
-            wcrstr = wcrstr[wcrstr.find('.') + 1:]  # Skip "ReductionType."
-
-        return 'Reduce ({op}), Axes: {axes}'.format(
-            axes=('all' if self.axes is None else str(self.axes)), op=wcrstr)
-
-    def __label__(self, sdfg, state):
-        return str(self).replace(' Axes', '\nAxes')
-
-    def validate(self, sdfg, state):
-        if len(state.in_edges(self)) != 1:
-            raise ValueError('Reduce node must have one input')
-        if len(state.out_edges(self)) != 1:
-            raise ValueError('Reduce node must have one output')
+# Copyright 2019-2021 ETH Zurich and the DaCe authors. All rights reserved.

```

Comment:
@tbennun There is no change in this file (reduce.py). It is marked as changed due to an issue with git and VS code on my WSL installation.